exec/java-exec/src/test/java/org/apache/drill/TestCTASJson.java

```diff
@@ -0,0 +1,129 @@
+   * Some records have missing fields. CTAS should skip the missing fields
+   */ public void testctas_alltypes_map() throws Exception {
+    String testName = "ctas_alltypes_map";
+    test("use dfs_test.tmp");
+    test("alter session set store.format = 'json' ");
+    test("alter session set store.json.writer.skip_null_fields = true"); // DEFAULT
+    test("create table " + testName + "_json as select * from cp.`json/" + testName + ".json`");
+
+    final String query = "select * from `" + testName + "_json` t1 ";
+
+    testBuilder()
+        .sqlQuery(query)
+        .ordered()
+        .jsonBaselineFile("json/" + testName + ".json")
+        .build()
+        .run();
+
+    test("drop table " + testName + "_json" );
+  }
+
+  @Test
+  /**
+   * Test a source json file that contains records that are maps with fields of all types.
+   * Some records have missing fields. CTAS should NOT skip the missing fields
+   */
+  public void testctas_alltypes_map_noskip() throws Exception {
+    String testName = "ctas_alltypes_map";
+    test("use dfs_test.tmp");
+    test("alter session set store.format = 'json' ");
+    test("alter session set store.json.writer.skip_null_fields = false"); // CHANGE from default
+    test("create table " + testName + "_json as select * from cp.`json/" + testName + ".json`");
+
+    final String query = "select * from `" + testName + "_json` t1 ";
+
+    testBuilder()
+        .sqlQuery(query)
+        .ordered()
+        .jsonBaselineFile("json/" + testName + "_out.json")
+        .build()
+        .run();
+
+    test("drop table " + testName + "_json" );
+  }
+
+  @Test
+  /**
+   * Test a source json file that contains records that are maps with fields of all types.
+   * Some records have missing fields. CTAS should skip the missing fields
+   */ public void testctas_alltypes_repeatedmap() throws Exception {
+    String testName = "ctas_alltypes_repeated_map";
+    test("use dfs_test.tmp");
+    test("alter session set store.format = 'json' ");
+    test("alter session set store.json.writer.skip_null_fields = true"); // DEFAULT
+    test("create table " + testName + "_json as select * from cp.`json/" + testName + ".json`");
+
+    final String query = "select * from `" + testName + "_json` t1 ";
+
+    testBuilder()
+        .sqlQuery(query)
+        .ordered()
+        .jsonBaselineFile("json/" + testName + ".json")
+        .build()
+        .run();
+
+    test("drop table " + testName + "_json" );
+
+  }
+
+  @Test
+  /**
+   * Test a source json file that contains records that are maps with fields of all types.
+   * Some records have missing fields. CTAS should NOT skip the missing fields
+   */
+  public void testctas_alltypes_repeated_map_noskip() throws Exception {
+    String testName = "ctas_alltypes_repeated_map";
+    test("use dfs_test.tmp");
+    test("alter session set store.format = 'json' ");
+    test("alter session set store.json.writer.skip_null_fields = false"); // CHANGE from default
```

Comment:
The testBuilder provides the optionSettingQueriesForTestQuery() method to set the options per query.  Would be good to use it and have a finally block {} that resets the option.